### ISSUE TITLE
Generates a open file error instead of a onecache.ErrCacheMiss

### ISSUE THREAD

**Issue by mikeschinkel** [adelowo/onecache#25]
https://github.com/adelowo/onecache/blob/ae397c7322a160a3f0e9ffba976417260bd732ec/filesystem/fs.go#L117

I was going to submit a pull request but could not figure out exactly how to fix it given some issues I was running into getting it set up on GoLand so I could both edit and debug it _(hard to edit and debug an external library in my IDE w/o renaming all the import paths.)_

Anyway, without this is makes error handling harder than it needs to be.

**Comment by adelowo** [comment]
Hi, Thanks... That error on the line is going to be ___an open file one___.  Kindly take a look at the Godoc of `os#Open` It is going to be an error of type `*PathError`.

Are you sure the item isn't expired ?

https://godoc.org/os#Open

**Comment by mikeschinkel** [comment]
What I was saying is that when I call `store.Get("profile")` from your README example that I do not expect to get a file open error after the cache has timed out. 

Actually, the first time I call it I get an `ErrCacheMiss` error, but the second time I call `store.Get()` I get a file open error. That is inconsistent and requires error handled to take into consideration both `ErrCacheMiss` and file open error when in reality what is really happening from the caller's perspective is a cache miss. 

The caller does not care that the cache file is no longer there, it is an implementation detail of your library.

Since I posted this I decided to write my own very simple cache library because it was taking me more time to get what I needed trying to figure out how to patch yours _(although [this article](https://splice.com/blog/contributing-open-source-git-repositories-go/) you linked me will help me in the future — thanks for that.)_ . And so I was able to implement the error handling code I am suggesting you consider implementing for your lines `116`-`118`:

```
if err != nil {
	pe, ok := err.(*os.PathError)
	if !ok {
		return nil,err
	}
	if pe.Err == syscall.ENOENT && pe.Op == "open" {
		return nil,ErrCacheMiss
	}
	return nil,err
}
```
